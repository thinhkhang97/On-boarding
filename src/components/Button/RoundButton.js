// @flow

import * as React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import Button from './Button';

type Props = {
  style?: View.propTypes.style,
  onPress: () => mixed,
  children: React.Node,
};

class RoundButton extends React.Component<Props> {
  static defaultProps = {
    style: {},
  };
  render() {
    const { onPress, style, children } = this.props;
    return (
      <Button
        style={[styles.container, style, styles.staticProps]}
        onPress={() => onPress && onPress()}
      >
        {children}
      </Button>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#1154ff',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: '#731154',
    shadowOffset: { width: 0, height: 5 },
    shadowRadius: 9,
    paddingLeft: 10,
    paddingRight: 10,
  },
  staticProps: {
    height: 44,
    borderRadius: 22,
    marginTop: 8,
    marginBottom: 8,
  },
});

export default RoundButton;
