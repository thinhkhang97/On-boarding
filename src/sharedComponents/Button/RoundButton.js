// Still get a trouble with shadow
import * as React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import Button from './Button';

type Props = {
  style?: object
};

class RoundButton extends React.Component<Props> {
  static defaultProps = {
    style: {}
  };
  render() {
    const { onPress } = this.props;
    return (
      <Button
        style={[
          styles.container,
          this.props.style,
          {
            height: 44,
            borderRadius: 22
          }
        ]}
        onPress={() => onPress && onPress()}
      >
        {this.props.children}
      </Button>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#1154ff',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    shadowColor: '#731154',
    shadowOffset: { width: 0, height: 5 },
    shadowRadius: 9,
    paddingLeft: 10,
    paddingRight: 10
  }
});

export default RoundButton;
