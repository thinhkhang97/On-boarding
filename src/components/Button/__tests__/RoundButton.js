import * as React from 'react';
import renderer from 'react-test-renderer';
import RoundButton from '../RoundButton';

test('Render round button correctly', ()=>{
    const testButton = renderer.create(<RoundButton/>).toJSON();
    expect(testButton).toMatchSnapshot();
})