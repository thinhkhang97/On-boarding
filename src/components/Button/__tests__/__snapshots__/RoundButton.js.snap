// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render round button correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1154ff",
      "borderRadius": 22,
      "display": "flex",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 8,
      "marginTop": 8,
      "opacity": 1,
      "paddingLeft": 10,
      "paddingRight": 10,
      "shadowColor": "#731154",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowRadius": 9,
    }
  }
/>
`;
